src/main/java/org/dita/dost/module/XsltModule.java

```diff
@@ -169,40 +151,62 @@ private void transform(final File in, final File out) throws DITAOTException {
             logger.debug("Set parameter " + filedirparameter + " to '" + v + "'");
             t.setParameter(new QName(filedirparameter), new XdmAtomicValue(v));
         }
+
+        if (properties.isEmpty()) {
```

Comment:
If serialization properties (encoding, indentation etc.) from `<xslt>` element is empty, we can defer the transformation to `Store`. This means `Store` decides how to serialize the files. This allows in-memory `Store` to just create the resources and replace them in the internal cache, instead of first creating them in the cache with a temporary resource URI and then replacing them into the final resource URI.